storage/queue.go

```diff
@@ -111,6 +112,96 @@ type PeekMessageResponse struct {
 	MessageText    string `xml:"MessageText"`
 }
 
+// QueueMetadata represents user defined metadata and queue properties on
+// a specific queue.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dd179384.aspx
+type QueueMetadata struct {
+	ApproximateMessageCount int
+	userDefinedMetadata     map[string]string
```

Comment:
no problem, but since Go's http client canonicalize the http header names, the (case-sensitive) map is not exactly what the user might expect. I would document this.